lib/atom-cfn-lint.js

```diff
@@ -66,90 +128,99 @@ export default {
-            return []
-          });
+        function emptyArray() {
+          return new Promise((resolve) => {
+            resolve([])
+          })
+        }
+
+        if (!(isCfn)) {
+          return emptyArray().then(() => [])
         }
 
         // parseable output is required
-        var args = ['--format', 'json']
+        let args = ['--format', 'json']
 
         // add file to check
-        args = args.concat(['--template', file]);
-        args = args.concat(['--ignore-bad-template']);
-
-        if (atom.config.getAll('atom-cfn-lint.ignoreRules').length > 0) {
-          args = args.concat(['--ignore-checks']);
-          for (i = 0; i < atom.config.get('atom-cfn-lint.ignoreRules').length; i++) {
-            args = args.concat([atom.config.get('atom-cfn-lint.ignoreRules')[i]]);
+        args = args.concat(['--template', file])
+        args = args.concat(['--ignore-bad-template'])
+
+        let i
+        if (this.ignoreRules) {
+          args = args.concat(['--ignore-checks'])
+          for (i = 0; i < this.ignoreRules.length; i += 1) {
+            args = args.concat([this.ignoreRules[i]])
           }
         }
 
-        if (atom.config.getAll('atom-cfn-lint.appendRules').length > 0) {
-          args = args.concat(['--append-rules']);
-          for (i = 0; i < atom.config.get('atom-cfn-lint.appendRules').length; i++) {
-            args = args.concat([atom.config.get('atom-cfn-lint.appendRules')[i]]);
+        if (this.appendRules) {
+          args = args.concat(['--append-rules'])
+          for (i = 0; i < this.appendRules.length; i += 1) {
+            args = args.concat([this.appendRules[i]])
           }
         }
 
-        if (atom.config.get('atom-cfn-lint.overrideSpecPath') !== '') {
-          args = args.concat(['--override-spec']);
-          args = args.concat([atom.config.get('atom-cfn-lint.overrideSpecPath')]);
+        if (this.overrideSpecPath) {
+          args = args.concat(['--override-spec'])
+          args = args.concat([this.overrideSpecPath])
         }
 
-        return helpers.exec(atom.config.get('atom-cfn-lint.cfnLintExecutablePath'), args, {cwd: require('path').dirname(file), ignoreExitCode: true}).then(output => {
-          JSON.parse(output).forEach(function (match) {
-            linenumber = parseInt(match.Location.Start.LineNumber) - 1;
-            columnnumber = parseInt(match.Location.Start.ColumnNumber) - 1;
-            linenumberend = parseInt(match.Location.End.LineNumber) - 1;
-            columnnumberend = parseInt(match.Location.End.ColumnNumber) - 1;
+        // initialize variable for linter return here for either linter output or errors
+        const toReturn = []
+
+        return helpers.exec(this.cfnLintExecutablePath, args, { cwd: require('path').dirname(file), ignoreExitCode: true }).then((output) => {
+          JSON.parse(output).forEach((match) => {
+            const linenumber = parseInt(match.Location.Start.LineNumber, 1000) - 1
```

Comment:
I _highly_ doubt that your line numbers are in base 1000. Check the docs of [parseInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt) 😉, especially since it only understands up to radix 36.